lib/parse.js

```diff
@@ -156,29 +166,51 @@ module.exports = {
    *
    * @param {Object} openapi OpenAPI
    * @param {Object} options options
+   * @param {Object} files Map of files path and content
    * @return {Object} Resolved content
    */
-  resolveContent: function (openapi, options) {
+  resolveContent: function (openapi, options, files) {
     return resolver.resolve(openapi, options.source, {
       options: Object.assign({}, options),
       resolve: true,
       cache: [],
       externals: [],
       externalRefs: {},
       rewriteRefs: true,
-      openapi: openapi
+      openapi: openapi,
+      files: files
     });
   },
 
   /** Resolves all OpenAPI file references and returns a single OAS Object
    *
    * @param {Object} source Root file path
    * @param {Object} options Configurable options as per oas-resolver module.
+   * @param {Object} files Map of files path and content
    * @return {Object} Resolved OpenAPI Schema
    */
-  mergeFiles: function(source, options = {}) {
+  mergeFiles: function(source, options = {}, files = {}) {
     options.source = source;
     options.origin = source;
+    if (!_.isEmpty(files)) {
+      let content = files[path.resolve(source)],
+        unresolved;
+      try {
+        unresolved = yamlParse.parse(content, { prettyErrors: true });
+      }
+      catch (err) {
+        throw new Error('\nLine: ' + err.linePos.start.line + ', col: ' +
```

Comment:
Where is this caught? What is the end-user experience like?

Reply:
This is caught in schemapack.mergeAndValidate. Currently in the app we aren't using this message anywhere. 
But could be something like: 
<img width="377" alt="Screenshot 2020-06-19 at 9 29 57 PM" src="https://user-images.githubusercontent.com/31915995/85160750-6184bf80-b27c-11ea-9a4b-d43979a5905c.png">
@abhijitkane